### PR TITLE
SDK changes for swagger update to support AutomaticRepairsPolicy for Virtual Machine Scale Sets

### PR DESCRIPTION
**Pull request by hapandar**
This SDK PR includes the below Swagger updates:

https://github.com/Azure/azure-rest-api-specs/pull/7484

https://github.com/Azure/azure-rest-api-specs/pull/7513

https://github.com/Azure/azure-rest-api-specs/pull/7355

https://github.com/Azure/azure-rest-api-specs/pull/7034

**Resolved. Including the additional changes**
Note: when I ran code gen tool, there were other changes (which were not part of my swagger changes) that were generated as well. Is the best practice to revert these changes? 
1) Since I am not familiar with the changes that were additionally generated and there may not be test coverage for this
2) There could already be other PRs in progress that would generate these additional properties along with tests. 
Please let me know and I can plan to revert those changes manually. For example, the ScaleInPolicy is not part of my changes and is not covered as part of the tests I have written.

### CODE REVIEW

File: sdk/compute/Microsoft.Azure.Management.Compute/src/Generated/Models/ScaleInPolicy.cs [hunk 1]
```diff
@@ -0,0 +1,92 @@
...
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License. See License.txt in the project root for
+// license information.
+//
+// Code generated by Microsoft (R) AutoRest Code Generator.
+// Changes may cause incorrect behavior and will be lost if the code is
+// regenerated.
+// </auto-generated>
+
+namespace Microsoft.Azure.Management.Compute.Models
```
Comment (hapandar) [inline sdk/compute/Microsoft.Azure.Management.Compute/src/Generated/Models/ScaleInPolicy.cs]: This generated file is not part of my swagger changes. Let me know if I should revert it.

File: sdk/compute/Microsoft.Azure.Management.Compute/src/Generated/Models/VirtualMachineScaleSet.cs [hunk 2]
```diff
@@ -78,15 +80,19 @@ public VirtualMachineScaleSet()
         /// Virtual Machine Scale Set. For instance: whether the Virtual
         /// Machines have the capability to support attaching managed data
         /// disks with UltraSSD_LRS storage account type.</param>
+        /// <param name="scaleInPolicy">Specifies the scale-in policy that
```
Comment (hapandar) [inline sdk/compute/Microsoft.Azure.Management.Compute/src/Generated/Models/VirtualMachineScaleSet.cs]: This generated property is not part of my swagger change.

File: sdk/compute/Microsoft.Azure.Management.Compute/src/Properties/AssemblyInfo.cs [hunk 3]
```diff
@@ -7,8 +7,8 @@
 [assembly: AssemblyTitle("Microsoft Azure Compute Management Library")]
 [assembly: AssemblyDescription("Provides management functionality for Microsoft Azure Compute Resources.")]
 
-[assembly: AssemblyVersion("29.1.0.0")]
-[assembly: AssemblyFileVersion("29.0.0.0")]
+[assembly: AssemblyVersion("29.2.0.0")]
+[assembly: AssemblyFileVersion("29.2.0.0")]
```
Comment (hyonholee) [inline sdk/compute/Microsoft.Azure.Management.Compute/src/Properties/AssemblyInfo.cs]: You don't have to update AssemblyFileVersion for minor version update.  Leave it to 29.0.0.0

File: sdk/compute/Microsoft.Azure.Management.Compute/src/Generated/Models/VirtualMachineScaleSet.cs (hunk 2 above)
Comment (hyonholee) [reply sdk/compute/Microsoft.Azure.Management.Compute/src/Generated/Models/VirtualMachineScaleSet.cs]: It was added by another previous Swagger PR.  You don't have to remove it.  Just leave it.

File: sdk/compute/Microsoft.Azure.Management.Compute/src/Microsoft.Azure.Management.Compute.csproj [hunk 4]
```diff
@@ -6,12 +6,12 @@
...
     <PackageId>Microsoft.Azure.Management.Compute</PackageId>
     <Description>Provides developers with libraries for the updated compute platform under Azure Resource manager to deploy virtual machine, virtual machine extensions and availability set management capabilities. Launch, restart, scale, capture and manage VMs, VM Extensions and more. Note: This client library is for Virtual Machines under Azure Resource Manager.</Description>
-    <Version>29.1.0</Version>
+    <Version>29.2.0</Version>
     <AssemblyName>Microsoft.Azure.Management.Compute</AssemblyName>
     <PackageTags>management;virtual machine;compute;</PackageTags>
     <PackageReleaseNotes>
       <![CDATA[
-        This is a public release of the Azure Compute SDK. Included with this release is adding $expand option for Availability Set List by Subscription API.
+        This is a public release of the Azure Compute SDK. Included with this release is adding Automatic Repairs Policy for Virtual Machine Scale Sets.
```
Comment (hyonholee) [inline sdk/compute/Microsoft.Azure.Management.Compute/src/Microsoft.Azure.Management.Compute.csproj]: Add more features by other Swagger updates.
- ScaleInPolicy for VMSS
- requireGeustProvisionSignal property in OsProfile.

File: sdk/compute/Microsoft.Azure.Management.Compute/src/Generated/Models/ScaleInPolicy.cs (hunk 1 above)
Comment (hyonholee) [reply sdk/compute/Microsoft.Azure.Management.Compute/src/Generated/Models/ScaleInPolicy.cs]: Do not revert it.  It was added by another PR.   Just leave it.

File: sdk/compute/Microsoft.Azure.Management.Compute/src/Properties/AssemblyInfo.cs (hunk 3 above)
Comment (hapandar) [reply sdk/compute/Microsoft.Azure.Management.Compute/src/Properties/AssemblyInfo.cs]: Fixed.

File: sdk/compute/Microsoft.Azure.Management.Compute/src/Microsoft.Azure.Management.Compute.csproj (hunk 4 above)
Comment (hapandar) [reply sdk/compute/Microsoft.Azure.Management.Compute/src/Microsoft.Azure.Management.Compute.csproj]: Updated.

**Comment by hyonholee** [comment]
LGTM

File: sdk/compute/Microsoft.Azure.Management.Compute/src/Properties/AssemblyInfo.cs (hunk 3 above)
Comment (isra-fel) [reply sdk/compute/Microsoft.Azure.Management.Compute/src/Properties/AssemblyInfo.cs]: @hyonholee @hapandar That is not correct. `AssemblyFileVersion` needs to be updated for every release, `AssemblyVersion` only when introducing a breaking change (aka a major version update).
So my suggestion is:
```suggestion
[assembly: AssemblyVersion("29.1.0.0")]
[assembly: AssemblyFileVersion("29.2.0.0")]
```
Read about it in our process document: https://github.com/Azure/adx-documentation-pr/blob/master/engineering/adx_netsdk_process.md#signing-and-publishing-nuget-packges
An article explaining this: https://support.microsoft.com/en-us/help/556041

**Review by isra-fel — CHANGES_REQUESTED** [review]
Please see my comment in the above conversation. Thanks

File: sdk/compute/Microsoft.Azure.Management.Compute/src/Properties/AssemblyInfo.cs (hunk 3 above)
Comment (hyonholee) [reply sdk/compute/Microsoft.Azure.Management.Compute/src/Properties/AssemblyInfo.cs]: @hapandar  Sorry, I was confused.  @isra-fel  is right.  Change only AssemblyFileVersion.

Comment (hapandar) [reply sdk/compute/Microsoft.Azure.Management.Compute/src/Properties/AssemblyInfo.cs]: Fixed

File: sdk/compute/Microsoft.Azure.Management.Compute/src/Microsoft.Azure.Management.Compute.csproj [hunk 5]
```diff
@@ -6,12 +6,17 @@
   <PropertyGroup>
     <PackageId>Microsoft.Azure.Management.Compute</PackageId>
     <Description>Provides developers with libraries for the updated compute platform under Azure Resource manager to deploy virtual machine, virtual machine extensions and availability set management capabilities. Launch, restart, scale, capture and manage VMs, VM Extensions and more. Note: This client library is for Virtual Machines under Azure Resource Manager.</Description>
-    <Version>29.1.0</Version>
+    <Version>29.2.0</Version>
```
Comment (hapandar) [inline sdk/compute/Microsoft.Azure.Management.Compute/src/Microsoft.Azure.Management.Compute.csproj]: Should I keep this version to 29.2.0 or change it back to 29.1.0.

Comment (hyonholee) [reply sdk/compute/Microsoft.Azure.Management.Compute/src/Microsoft.Azure.Management.Compute.csproj]: @hapandar I updated this one for minor version update in the past.  Is it not necessary?

Comment (isra-fel) [reply sdk/compute/Microsoft.Azure.Management.Compute/src/Microsoft.Azure.Management.Compute.csproj]: `29.2.0` is the correct version. A simple rule:
> Nuget Package version change (in .csproj), AssemblyFileVersion change. If it's a major version update, AssemblyVersion will also need to be changed (in AssemblyInfo.cs)

**Review by isra-fel — APPROVED** [review]

**Pull request merged**